cache.go

```diff
@@ -5,17 +5,26 @@ import (
 	"time"
 )
 
+// CachedValue is a type for cache value
+type CachedValue interface{}
```

Comment:
I like this idea - what do you think of calling this type `T` instead of `CachedValue`? 
I'm hoping that would be clearer to users what that type is meant to represent.